composer/algorithms/seq_length_warmup/seq_length_warmup.py

```diff
@@ -306,27 +307,25 @@ def _activate_model(self, state: State, logger: Logger) -> None:
             # This error/state.grad_accum handling mimics the logic in trainer._train_batch().
             except RuntimeError as e:
                 if 'CUDA out of memory' in str(e):
-                    should_handle_cuda_oom = 1
+                    found_cuda_oom = 1
                 else:
                     raise
 
-            # Propagate across all ranks if any rank hit CUDA OOM
-            should_handle_cuda_oom = torch.tensor([should_handle_cuda_oom], dtype=torch.uint8, device=device)
-            dist.all_reduce(should_handle_cuda_oom, reduce_operation='MAX')
-            if int(should_handle_cuda_oom.item()) == 1:
-                # If any rank hit CUDA OOM, update grad_accum and retry. Ignore any caught_timeout_error since
-                # it is likely transient, e.g. timeout because certain ranks OOMed and didn't reach barrier.
-                # Raise runtime error if training 1 sample at a time still resulted in CUDA out of memory
-                if state.grad_accum == device_batch_size:
-                    raise RuntimeError(
-                        ('CUDA out of memory. The train loop failed with an internal microbatch of size 1.'
-                         'The GPU does not have enough memory to process even 1 sample.'))
-                else:
-                    state.grad_accum = min(2 * state.grad_accum, device_batch_size)
-                    logger.data_batch({'trainer/grad_accum': state.grad_accum})
-            else:
-                self._activated = True
-                return
+            # In-line to avoid circular dependency
+            from composer.trainer.trainer import _handle_train_cuda_oom
+
+            # Auto grad accum only supported on GPU
+            if device and device.type == 'gpu':
+                devicegpu = DeviceGPU()
+                # Propagate across all ranks if any rank hit CUDA OOM
+                found_cuda_oom = devicegpu.tensor_to_device(torch.tensor([found_cuda_oom], dtype=torch.uint8)).item()
+                if found_cuda_oom == 1:
+                    _handle_train_cuda_oom(state, device_batch_size)
```

Comment:
personal preference, so feel free to ignore, but prefer not obfuscating the side effect and being more transparent on what is happening, e.g. `_adjust_grad_accum`. otherwise, the reader needs to check another function to actually understand what is happening.